aiidalab_widgets_base/process.py

```diff
@@ -111,41 +115,111 @@ def on_submitted(self, function):
+            if change['new']:
+                selected_input = self.process.inputs[change['new']]
+                self.info.value = "PK: {}".format(selected_input.id)
+                display(viewer(selected_input))
+
+
+class ProcessOutputsWidget(ipw.VBox):
+    """Widget to select and show process outputs."""
+    process = Instance(ProcessNode, allow_none=True)
+
+    def __init__(self, process=None, **kwargs):
+        self.process = process
+        self.output = ipw.Output()
+        self.info = ipw.HTML()
+        outputs_list = [(l.title(), l) for l in self.process.outputs] if self.process else []
+        outputs = ipw.Dropdown(
+            options=[('Select output', '')] + outputs_list,
+            label='Select output',
+            description='Select outputs:',
+            style={'description_width': 'initial'},
+            disabled=False,
+        )
+        outputs.observe(self.show_selected_output, names=['value'])
+        super().__init__(children=[ipw.HBox([outputs, self.info]), self.output], **kwargs)
+
+    def show_selected_output(self, change=None):
+        """Function that displays process output selected in the `outputs` Dropdown widget."""
+        with self.output:
+            self.info.value = ''
+            clear_output()
+            if change['new']:
+                selected_output = self.process.outputs[change['new']]
+                self.info.value = "PK: {}".format(selected_output.id)
+                display(viewer(selected_output))
+
+
+class ProcessFollowerWidget(ipw.VBox):
     """A Widget that follows a process until finished."""
+    process = Instance(ProcessNode, allow_none=True)
 
-    def __init__(self, process, followers=None, update_interval=0.1, **kwargs):
+    def __init__(self, process=None, followers=None, update_interval=0.1, path_to_root='../', **kwargs):
         """Initiate all the followers."""
-        if not isinstance(process, ProcessNode):
-            raise TypeError("Expecting an object of type {}, got {}".format(ProcessNode, type(process)))
         self.process = process
         self._run_after_completed = []
         self.update_interval = update_interval
         self.followers = []
         if followers is not None:
             for follower in followers:
-                self.followers.append(follower(process=process))
+                follower.process = self.process
+                follower.path_to_root = path_to_root
+                self.followers.append(ipw.VBox([
+                    ipw.HTML("<h2><b>{}</b></h2>".format(follower.title)),
+                    follower,
+                ]))
         self.update()
-        super(ProcessFollowerWidget, self).__init__(children=self.followers, **kwargs)
+        self.output = ipw.HTML()
+        super().__init__(children=[self.output] + self.followers, **kwargs)
 
     def update(self):
         for follower in self.followers:
-            follower.update()
+            follower.children[1].update()
```

Comment:
Is this to be recursive? Wouldn't it need to be something like `for child in follower.children[1:]: child.update()`?

Reply:
No, it is not recursive. It is because I am appending follower together with the title in one `VBox` block.
```
                self.followers.append(ipw.VBox([
                    ipw.HTML("<h2><b>{}</b></h2>".format(follower.title)),
                    follower,
                ]))
```

so to access the follower I use the index.